parsons/etl_helpers/conversions.py

```diff
@@ -0,0 +1,188 @@
+
+
+# Flatten contacts
+
+def get_primary_contact_from_nested(contact_list, get_first=True, selector=None):
+    """Extracts single contact value from list of dictionaries.
+
+    Multiple connectors, including Mobilize and Action Network, have a format for
+    contact info that looks like:
+
+    [{"primary": True, "number": "111-111-1111"}]
+
+    This helper method helps "flatten" the dictionary by returning the primary number (or,
+    if no primary number is found and get_first is True, the first number found).
+
+    If no selector is passed in, the function tries common options and, finally, tries to get the
+    selector from the dictionary.
+
+    `Args:`
+        contact_list: list
+            List containing dictionaries with format {"primary": True, $selector: $value}
+        get_first: boolean
+            `Optional`: If primary contact is not found, return first contact if it exists rather
+            than None; defaults to True
+        selector: str
+            `Optional`:
+    `Returns:`
+        str
+    """
+
+    if not contact_list:
+        return
+
+    # get selector if not supplied
+
+    if not selector:
+
+        for option in ["number", "postal_code", "address"]:  # try common options
+            if option in contact_list[0]:
+                selector = option
+                break
+
+        if not selector:  # if selector still not found, look in dict
+            dict_keys = list(contact_list[0].keys())
+            dict_keys.pop("primary")
+            selector = dict_keys[0]  # NOTE: this will break on dicts that have additional keys
+
+    # extract contact using selector
+
+    extracted_contacts = []
+
+    for contact in contact_list:
+
+        if contact["primary"]:
+            return contact[selector]
+        else:
+            extracted_contacts.append(contact[selector])
+
+    if get_first and extracted_contacts:
+        return extracted_contacts[0]
+
+    return None
+
+
+def clean_US_phones(value, simple=True):
+    """Provides simple validation and standardization for US phone numbers.
+
+    Given a phone number, standardizes to either a simple format or a fancy format.
+
+    Simple: 10 digit format with no country code, spaces or extra characters, ie: 3334445555
+    Fancy: 16 digit format with parens and country code, ie: 1-(333)-444-5555
+
+    Assumes North American phone number and will incorrectly reject most international numbers.
+
+    `Args:`
+        value: str
+            Value containing phone number to parse
+        simple: boolean
+            `Optional`: Indicates format phone number should be standardized to; defaults to True
+    `Returns:`
+        str
+    """
+
+    # Parse string and standardize to 10 digit format
+
+    value = value.split("x")[0]  # if there's an x in the old_number, use only numbers before the x
+
+    parsed_value = re.compile(r'\d+(?:\.\d+)?').findall(value)  # extracts digits only
+    value = "".join(parsed_value)
```

Comment:
I find this a bit more readable / easier to maintain (regexes, in my view, should be avoided at all costs): 

```suggestion
    value = "".join([v for v in value if v.isdigit()])
```